fix: block Google Meet installation without Google Calendar dependency

pedroccastro opened the pull request:
## What does this PR do?

This PR fixes a UX issue where users could install Google Meet without having Google Calendar connected, resulting in a broken experience where Google Meet would silently fallback to Daily.co video.

  **Key Changes:**
Block Google Meet installation when Google Calendar dependency is not met
Update visual feedback to show error state (red background + X icon) instead of info state
Fix button component to properly respect disabled state from dependency checks
Ensure consistent behavior between App Store listing and app detail pages

  - Fixes #25497
  - Linear [CAL-6846](https://linear.app/calcom/issue/CAL-6846/bug-api-generates-dailycal-video-link-even-when-google-meet-is)

  ## Visual Demo

  #### Image Demo:
 
 **Before:**

<img width="616" height="282" alt="before_warning" src="https://github.com/user-attachments/assets/bd6288db-3d85-4af5-a43d-99b47d170693" />

  **After:**

<img width="613" height="282" alt="after_blocking" src="https://github.com/user-attachments/assets/ea1803e0-8f33-4425-b48b-a00be27edc79" />


**Screenshots showing:**

  1. Google Meet app page with Google Calendar not installed (error state)
  2. Disabled "Install App" button
  3. Clear error message: "Google Meet requires a connected Google Calendar account"
  4. Working flow after installing Google Calendar first

  ## Mandatory Tasks (DO NOT REMOVE)

  - [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
  - [x] **N/A** I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). 
  - [ ] I confirm automated tests are in place that prove my fix is effective or that my feature works.

  ## How should this be tested?

  **Test Scenario 1 - Google Meet blocked without Calendar:**
  1. Ensure Google Calendar app is NOT installed
  2. Navigate to `/apps/google-meet`
  3. Verify "Install App" button is disabled
  4. Verify red error message appears: "Google Meet requires a connected Google Calendar account"
  5. Verify "Connect Google Calendar" link works

  **Test Scenario 2 - Google Meet allowed with Calendar:**
  1. Install Google Calendar app first
  2. Navigate to `/apps/google-meet`
  3. Verify green checkmark shows "Google Calendar is connected"
  4. Verify "Install App" button is enabled
  5. Verify Google Meet installs successfully

  **Test Scenario 3 - App Store consistency:**
  1. Check `/apps` listing page
  2. Verify Google Meet shows same disabled state when Calendar not installed
  3. Verify consistency between listing and detail pages

  **Expected:** Google Meet cannot be installed without Google Calendar dependency

pedroccastro committed:
fix: block Google Meet installation without Google Calendar dependency

  - Prevent users from installing Google Meet when Google Calendar is not connected
  - Update AppDependencyComponent to show error state for unmet dependencies
  - Fix InstallAppButtonChild to respect disabled prop from dependencies check
  - Ensure consistent dependency validation across App Store and app detail pages

  Fixes #25497

pedroccastro committed:
Merge branch 'main' into fix/block-google-meet-without-calendar

pedroccastro committed:
test: add unit tests for dependency validation
- Add tests for AppDependencyComponent (8 tests)
- Add tests for InstallAppButtonChild (5 tests)
- Cover dependency blocking behavior
- Test visual state changes (bg-error/bg-subtle)

pedroccastro committed:
Merge branch 'fix/block-google-meet-without-calendar' of https://github.com/calcom/cal.com into fix/block-google-meet-without-calendar

pedroccastro committed:
test: resolve type-check error in InstallAppButtonChild tests

- Add complete MockCredential type definition
- Create factory function to generate typed mock credentials
- Eliminate code duplication across test cases

hariombalhara commented on apps/web/components/apps/AppPage.tsx:
```diff
@@ -230,6 +229,7 @@ export const AppPage = ({
                 handleAppInstall();
               },
               loading: isLoading,
+              disabled: disableInstall || props.disabled,
```
[It seems like _disableInstall_ is translated to _props.disabled_](https://github.com/calcom/cal.com/blob/12-03-feat_add_event_handling_for_availability_and_booking_events/packages/app-store/InstallAppButtonWithoutPlanCheck.tsx#L25) and thus we don't need to use disableInstall variable from outside the component. 

We need to figure out what case is there where disableInstall isn't mapped to props.disabled

hariombalhara commented on apps/web/components/apps/AppPage.tsx:
```diff
@@ -250,6 +250,7 @@ export const AppPage = ({
                 handleAppInstall();
               },
               loading: isLoading,
+              disabled: disableInstall || props.disabled,
```
Same here

hariombalhara reviewed (changes requested):
https://github.com/calcom/cal.com/pull/25513/files#r2591378313
https://github.com/calcom/cal.com/pull/25513/files#r2591378388

We can avoid these unnecessary changes, otherwise it is good

pedroccastro committed:
fix: remove redundant disabled prop assignment

Disabled state is already passed through via props.disableInstall
which gets converted to props.disabled in InstallAppButtonWithoutPlanCheck.
This change addresses the review comment to avoid unnecessary duplication

pedroccastro committed:
Merge branch 'main' into fix/block-google-meet-without-calendar

Udit-takkar committed:
Merge branch 'main' into fix/block-google-meet-without-calendar

Udit-takkar reviewed (approved):
LGTM
